### ISSUE TITLE
allow the energy-per-pulse (EPP) to be set when the pulse is created

### ISSUE THREAD

**Issue by DanHickstein** [DanHickstein/pyNLSE#12]
currently it takes two lines to create the pulse: 

```python
# create the pulse
pulse = nlse.pulse.SechPulse(power=1, T0_ps=FWHM/1.76, center_wavelength_nm=pulseWL, 
                             time_window_ps=Window, GDD=GDD, TOD=TOD, NPTS= Points, 
                             frep_MHz=100, power_is_avg=False)

pulse.set_epp(EPP)  # set the pulse energy
```

we should allow either the power or the pulse energy to be set.

**Comment by DanHickstein** [comment]
This is going to be tedious because it needs to be done to every pulse creation function (Gaussian, Sech, etc.).

Actually, maybe it's better to combine all of the pulse generation functions into one "generatePulse" function with a kw argument for the type of pulse to be generated. This would certainly reduce the boiler-plate code in pulse.py...

**Comment by DanHickstein** [comment]
This was implemented in #18
